### PR TITLE
[Docs] Transformers DeepSparse Integration (also a template for the future similar, docs)

### PR DESCRIPTION
**Pull request by dbogunowicz**
Todos:

- [x] let's include some mention of model sparsification (sparsity and quantization) and DeepSparse being specialized for these types of models

- [x] in the getting started for ONNX Model Support let's rename this to something more generic such as Model Format. Underneath let's better organize everything we need in a local folder including the model.onnx, the tokenizer, etc. Can then call out that a SparseZoo stub can also be supplied and it will download all necessary files

- [x] Also underneath this let's link out to export documentation in SparseML for the user to dive in further

- [ ] Let's remove the duplicate example of question answering for Python deployment. Instead, let's call out an API reference at the top and then give example code implementation for the different use cases

- [x] Let's format the server in the same way as above for the Python deployment. Start with a CLI reference for the server and command args and then give examples for the different use cases

### CODE REVIEW

**Review by markurtz — CHANGES_REQUESTED** [review]
Quick feedback on the DeepSparse template:

- let's include some mention of model sparsification (sparsity and quantization) and DeepSparse being specialized for these types of models
- in the getting started for ONNX Model Support let's rename this to something more generic such as Model Format. Underneath let's better organize everything we need in a local folder including the model.onnx, the tokenizer, etc. Can then call out that a SparseZoo stub can also be supplied and it will download all necessary files
- Also underneath this let's link out to export documentation in SparseML for the user to dive in further
- Let's remove the duplicate example of question answering for Python deployment. Instead, let's call out an API reference at the top and then give example code implementation for the different use cases
- Let's format the server in the same way as above for the Python deployment. Start with a CLI reference for the server and command args and then give examples for the different use cases

**Comment by dbogunowicz** [comment]
@markurtz thank you for your comments, addressed most of them. That being said, could you clarify what you mean by:

```Let's remove the duplicate example of question answering for Python deployment. Instead, let's call out an API reference at the top and then give an example code implementation for the different use cases```

I am confused about what you mean by the API reference here. API reference for the `transformer.pipeline`? Is there a nice "user-facing" API reference for that?

File: src/deepsparse/transformers/README.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+As indicated above, `model_path` may be a path to a local model directory, however, in the examples below, we set the `model_path` argument to the model stub of our SparseZoo models. 
+
+
+#### Question Answering Pipeline
+
+[List of the Hugging Face SparseZoo Question Answering Models](
+https://sparsezoo.neuralmagic.com/?page=1&domain=nlp&sub_domain=question_answering)
+
+```python
+from deepsparse.transformers import pipeline
```
[the remark this answers is not in the document]
Comment (dbogunowicz) [reply src/deepsparse/transformers/README.md:107]: Interesting. Is this the intended behavior now? I don't see it working with the current `main`:

```
Type "help", "copyright", "credits" or "license" for more information.
>>> from deepsparse import pipeline
>>> pipeline
<module 'deepsparse.pipeline' from '/home/damian/deepsparse_copy/src/deepsparse/pipeline.py'>
>>> from deepsparse.transformers import pipeline
>>> pipeline
<function pipeline at 0x7f2d0b3ead30>
```

Let's discuss this internally.

File: src/deepsparse/transformers/README.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+qa_pipeline = pipeline(
+    task="question-answering",
+    model_path=model_path)
+
+inference = qa_pipeline(question="What's my name?", context="My name is Snorlax")
+
+>> {'score': 0.9947717785835266, 'start': 11, 'end': 18, 'answer': 'Snorlax'}
+```
+
+#### Text Classification Pipeline
```
Comment (markurtz) [inline src/deepsparse/transformers/README.md:120]: Let's call out sentiment analysis as a separate pipeline and use a text classification example for mnli or qqp

File: src/deepsparse/transformers/README.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+>>True
+```
+
+
+## Deployment
+
+### Python API
+Python API is the default interface for running the inference with the DeepSparse Engine.
+
+The SparseML installation provides a CLI for sparsifying your models for a specific task; appending the `--help` argument displays a full list of options for training in SparseML:
```
[the remark this answers is not in the document]
Comment (bfineran) [reply src/deepsparse/transformers/README.md:80]: fixing - will add in links after final filenames are set in sparseml

**Comment by bfineran** [comment]
@markurtz refactored server section to point to the server docs

File: src/deepsparse/transformers/README.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
+# Hugging Face Transformer Inference Pipelines
+
+
+DeepSparse allows accelerated inference, serving, and benchmarking of sparsified [HuggingFace transformer](https://github.com/huggingface/transformers) models.
```
Comment (jeanniefinks) [inline src/deepsparse/transformers/README.md:4]:
```suggestion
DeepSparse allows accelerated inference, serving, and benchmarking of sparsified [Hugging Face Transformer](https://github.com/huggingface/transformers) models.  
```

File: src/deepsparse/transformers/README.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
+# Hugging Face Transformer Inference Pipelines
+
+
+DeepSparse allows accelerated inference, serving, and benchmarking of sparsified [HuggingFace transformer](https://github.com/huggingface/transformers) models.  
+This integration allows for leveraging the DeepSparse Engine to run the sparsified transformer inference with GPU-class performance directly on the CPU.
+
+The DeepSparse Engine is taking advantage of sparsity within neural networks to 
+reduce compute required as well as accelerate memory-bound workloads. The Engine is particularly effective when leveraging sparsification
```
Comment (jeanniefinks) [inline src/deepsparse/transformers/README.md:8]:
```suggestion
reduce compute required as well as accelerate memory-bound workloads. The engine is particularly effective when leveraging sparsification
```

File: src/deepsparse/transformers/README.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+Before you start your adventure with the DeepSparse Engine, make sure that your machine is 
+compatible with our [hardware requirements](https://docs.neuralmagic.com/deepsparse/source/hardware.html).
+
+### Installation
+
+```pip install deepsparse```
+
+### Model Format
+By default, to deploy the transformer using DeepSparse Engine it is required to supply the model in the ONNX format. 
+This grants the Engine the flexibility to serve any model in a framework-agnostic environment.
```
Comment (jeanniefinks) [inline src/deepsparse/transformers/README.md:31]:
```suggestion
This grants the engine the flexibility to serve any model in a framework-agnostic environment. 
```

File: src/deepsparse/transformers/README.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+
+```pip install deepsparse```
+
+### Model Format
+By default, to deploy the transformer using DeepSparse Engine it is required to supply the model in the ONNX format. 
+This grants the Engine the flexibility to serve any model in a framework-agnostic environment. 
+
+Below we describe two possibilities to obtain the required ONNX model.
+
+#### Exporting the onnx file from the contents of a local directory
```
Comment (jeanniefinks) [inline src/deepsparse/transformers/README.md:35]:
```suggestion
#### Exporting the ONNX file from the contents of a local directory
```

File: src/deepsparse/transformers/README.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+This pathway is relevant if you intend to deploy a model created using [SparseML](https://github.com/neuralmagic/sparseml) library. 
+For more information, refer to the appropriate transformers integration documentation in SparseML.
+
+The expected `model_path` in a transformers `Pipeline` should be a directory that includes the following files:
+ - `model.onnx`
+ - `tokenizer.json`
+ - `config.json`
+
+Onnx models can be exported using the `sparseml.transformers.export_onnx` tool:
```
Comment (jeanniefinks) [inline src/deepsparse/transformers/README.md:44]:
```suggestion
ONNX models can be exported using the `sparseml.transformers.export_onnx` tool:
```

File: src/deepsparse/transformers/README.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+
+```bash
+sparseml.transformers.export_onnx --task question-answering --model_path model_path
+```
+This creates `model.onnx` file, in the parent directory of your `model_path`(e.g. `/trained_model/model.onnx`)
+
+####  Directly using the SparseZoo stub
+Alternatively, you can skip the process of onnx model export by downloading all the required model data directly from Neural Magic's [SparseZoo](https://sparsezoo.neuralmagic.com/).
```
Comment (jeanniefinks) [inline src/deepsparse/transformers/README.md:52]:
```suggestion
Alternatively, you can skip the process of ONNX model export by downloading all the required model data directly from Neural Magic's [SparseZoo](https://sparsezoo.neuralmagic.com/).
```

File: src/deepsparse/transformers/README.md [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+    {'entity': 'LABEL_5','word': 'texas', ...}, 
+    {'entity': 'LABEL_0','word': '!', ...}]
+```
+
+### DeepSparse Server
+As an alternative to Python API, the DeepSparse inference server allows you to serve ONNX models and pipelines in HTTP.
+Configs for the server support the same arguments as the above pipelines and setting the task and models to any of the
+above transformers tasks and models will enable easy deployment.
+
+For a full example of deploying sparse transformer models with the DeepSparse server, see the
```
Comment (jeanniefinks) [inline src/deepsparse/transformers/README.md:158]:
```suggestion
For a full example of deploying sparse transformer models with the DeepSparse Server, see the
```

File: src/deepsparse/transformers/README.md [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+>> [{'entity': 'LABEL_0','word': 'drive', ...}, 
+    {'entity': 'LABEL_0','word': 'from', ...}, 
+    {'entity': 'LABEL_5','word': 'california', ...}, 
+    {'entity': 'LABEL_0','word': 'to', ...}, 
+    {'entity': 'LABEL_5','word': 'texas', ...}, 
+    {'entity': 'LABEL_0','word': '!', ...}]
+```
+
+### DeepSparse Server
+As an alternative to Python API, the DeepSparse inference server allows you to serve ONNX models and pipelines in HTTP.
```
Comment (jeanniefinks) [inline src/deepsparse/transformers/README.md:154]:
```suggestion
As an alternative to Python API, the DeepSparse Server allows you to serve ONNX models and pipelines in HTTP.
```

**Review by jeanniefinks — CHANGES_REQUESTED** [review]
Few minor edits

File: src/deepsparse/transformers/README.md [hunk 12]
```diff
@@ -0,0 +1,324 @@
...
+
+### Python Pipelines
+Pipelines are the default interface for running inference with the DeepSparse Engine.
+
+Once a model is obtained, either through `SparseML` training or directly from `SparseZoo`,
+`deepsparse.Pipeline` can be used to easily facilitate end to end inference and deployment
+of the sparsified transformers model.
+
+If no model is specified to the `Pipeline` for a given task, the `Pipeline` will automatically
+select a pruend and quantized model for the task from the `SparseZoo` that can be used for accelerated
```
Comment (KSGulin) [inline src/deepsparse/transformers/README.md:72]:
```suggestion
select a pruned and quantized model for the task from the `SparseZoo` that can be used for accelerated
```

**Review by markurtz — APPROVED** [review]

**Pull request merged**